backend/auth/role_map_repository.go

```diff
@@ -393,36 +127,97 @@ func GetRoleMapConfig(namespace string, name string) (map[string]*models.Role, m
 
 	details := (*res.ResourceDetails).(*unstructured.Unstructured)
 
-	roleMapData, foundRoleMap, err2 := unstructured.NestedString(details.Object, "data", "role-map")
-	if err2 != nil || !foundRoleMap {
+	roleMapConfigData, foundRoleMapConfig, err2 := unstructured.NestedString(details.Object, "data", "role-map")
+	if err2 != nil || !foundRoleMapConfig {
 		log.Printf("Error retrieving roleMap data: %v", err)
 		return nil, nil
 	}
```

Comment:
Tell me what you think and if this would work.

```
func getConfigMapData(details *unstructured.Unstructured, key string) (map[string]*roleConfig, error) {
    data, found, err := unstructured.NestedString(details.Object, "data", key)
    if err != nil || !found {
        return nil, fmt.Errorf("error retrieving %s data: %w", key, err)
    }

    config := make(map[string]*roleConfig)
    if err := yaml.Unmarshal([]byte(data), &config); err != nil {
        return nil, fmt.Errorf("error parsing %s data: %w", key, err)
    }
    return config, nil
}
```

```
roleMapConfig, err := getConfigMapData(details, "role-map")
if err != nil {
    return nil, nil, err
}

subroleMapConfig, err := getConfigMapData(details, "subrole-map")
if err != nil {
    log.Printf("%v", err)
} else {
    subroleMap = fromRoleMapConfig(subroleMapConfig)
    // ... rest of the code ...
}
```

Reply:
I am not a fan of this refactor cause im lazy